ci-operator/step-registry/cucushift/agent/vsphere/day2/add-nodes/cucushift-agent-vsphere-day2-add-nodes-commands.sh

```diff
@@ -0,0 +1,248 @@
+
+folder_name=$(<"${SHARED_DIR}"/cluster-name.txt)
+total_workers="${ADDITIONAL_WORKERS_DAY2}"
+declare -a mac_addresses=()
+
+for ((i = 0; i < total_workers; i++)); do
+  mac_addresses+=(00:50:56:ac:b8:1"$i")
+  echo "${mac_addresses[$i]}"
+done
+
+declare -a hostnames=()
+for ((i = 0; i < total_workers; i++)); do
+  hostnames+=("${folder_name}-additional-worker-$i")
+  echo "${hostnames[$i]}"
+done
+
+for ((i = 0; i < total_workers; i++)); do
+  ipaddress=$(jq -r --argjson N $((i + 10)) --arg PRH "$primaryrouterhostname" --arg VLANID "$vlanid" '.[$PRH][$VLANID].ipAddresses[$N]' "${SUBNETS_CONFIG}")
+  ipv6_address=$(jq -r --arg PRH "$primaryrouterhostname" --arg VLANID "$vlanid" '.[$PRH][$VLANID].StartIPv6Address' "${SUBNETS_CONFIG}")
+
+  ipv4="
+        ipv4:
+          enabled: true
+          address:
+            - ip: ${ipaddress}
+              prefix-length: ${cidr}
+          dhcp: false"
+  ipv6="
+        ipv6:
+          enabled: true
+          address:
+            - ip: "${ipv6_address%%::*}::"$((i + 12))
+              prefix-length: ${cidr_ipv6}
+          dhcp: false"
+  route_ipv4="
+          - destination: 0.0.0.0/0
+            next-hop-address: ${gateway}
+            next-hop-interface: ens32
+            table-id: 254"
+  route_ipv6="
+          - destination: ::/0
+            next-hop-address: ${gateway_ipv6}
+            next-hop-interface: ens32
+            table-id: 254"
+  # Single-stack override conditions
+  if [[ ${IP_FAMILIES} == "IPv4" ]]; then
+    ipv6=""
+    route_ipv6=""
+  fi
+  if [[ ${IP_FAMILIES} == "IPv6" ]]; then
+    ipv4=""
+    route_ipv4=""
+  fi
+  echo " - hostname: ${hostnames[$i]}
+   role: $(echo "${hostnames[$i]}" | rev | cut -d'-' -f2 | rev | cut -f1)
+   interfaces:
+    - name: ens32
+      macAddress: ${mac_addresses[$i]}
+   networkConfig:
+    interfaces:
+      - name: ens32
+        type: ethernet
+        state: up
+        mac-address: ${mac_addresses[$i]}${ipv4}${ipv6}
+    dns-resolver:
+     config:
+      server:
+       - ${dns_server}
+    routes:
+     config:${route_ipv4}${route_ipv6}"
+done >>"${SHARED_DIR}/nodes-config.yaml.patch"
+
+nodes_config_patch="${SHARED_DIR}/nodes-config.yaml.patch"
+#create agent node config file
+cat >"${SHARED_DIR}/nodes-config.yaml" <<EOF
+hosts: []
+EOF
+
+nodes_config="${SHARED_DIR}/nodes-config.yaml"
+#Add hosts details to the nodes-config.yaml
+yq-v4 --inplace eval-all 'select(fileIndex == 0).hosts += select(fileIndex == 1) | select(fileIndex == 0)' \
+  "${nodes_config}" - <<<"$(cat "${nodes_config_patch}")"
+
+echo "Creating agent node image..."
+dir=/tmp/installer
+mkdir "${dir}/"
+pushd ${dir}
+cp -t "${dir}" "${SHARED_DIR}"/nodes-config.yaml
+
+export KUBECONFIG="${SHARED_DIR}/kubeconfig"
+
+curl https://raw.githubusercontent.com/openshift/installer/master/docs/user/agent/add-node/node-joiner.sh --output "${dir}/node-joiner.sh"
+
+chmod +x "${dir}/node-joiner.sh"
+sh "${dir}/node-joiner.sh"
```

Comment:
@mhanss, I think it would be better to catch the exit status of the `node-joiner` here to confirm that everything worked as expected, WDYT?

BTW, the aforementioned script has no specific exit codes, I can only see this part as reliable status source:
~~~
res=$(oc exec node-joiner -n "${namespace}" -- cat /assets/exit_code)
if [ "$res" = 0 ]; then
  echo "node-joiner successfully completed, extracting ISO image..."
  oc cp -n "${namespace}" node-joiner:/assets/node.x86_64.iso node.x86_64.iso
else
  oc logs node-joiner -n "${namespace}"
  echo "node-joiner failed"
fi
~~~

Reply:
Yes, we can catch the exit status here, but the next command already ensures the file exists during the upload and will return an error if the file does not exist, something like this:
`govc: stat node.x86_64.iso: no such file or directory`
Do you think we should still add some code to verify the existence of the file?

Reply:
good point, I think the file existance check could be sufficient in this case, thanks.